### ISSUE TITLE
Morphdom Exception: component.js:257 TypeError: Cannot read properties of undefined (reading 'nodeType')

### ISSUE THREAD

**Issue by winrid** [adamghill/django-unicorn#631]
Component only works until parent component removes it and adds a different instance with a different model... only happens when used as child component. just get this any time I change the `<select>` model value via the select:

```
component.js:257 TypeError: Cannot read properties of undefined (reading 'nodeType')
    at morphdom (morphdom.js:727:29)
    at MorphdomMorpher.morph (morphdom.js:9:12)
    at messageSender.js:229:27
```

Currently reproducible via sidewaysdata user dashboard. Just change "type" dropdown to any value. Happens any time dropdown value changes. Cards get added/removed from DOM still fine with this error. However, once you re-render the whole component, it stops working.

Again - only happens when a child component. When used directly from Django view I do not see this exception.

**Comment by winrid** [comment]
@adamghill Here is a video:

https://youtu.be/pDZQa87mU4Y

I tried to fix this in sideways data commit cb98d2c. So basically all I have is:

```
{% unicorn 'find_events' user=user filter_event_ids=selected_event_ids %}
```

Then I just change `selected_event_ids` via:

```
    def switch_tab(self, new_tab: str):
        self.current_tab = new_tab
        if self.current_tab == 'competing':
            self.selected_event_ids = self.competitor_upcoming_event_ids
        elif self.current_tab == 'organizing':
            self.selected_event_ids = self.my_organized_event_ids
        elif self.current_tab == 'volunteering':
            self.selected_event_ids = self.upcoming_volunteering_event_ids
        elif self.current_tab == 'volunteer-captain':
            self.selected_event_ids = self.upcoming_volunteering_captain_event_ids
        elif self.current_tab == 'completed':
            self.selected_event_ids = self.competitor_entries_completed_event_ids
        else:
            self.selected_event_ids = []
```

But this doesn't work.

So I have no way to have multiple children in the same component? I am very confused. It's pretty frustrating.

The exception in the ticket and the child component might working may or may not be related...

**Comment by winrid** [comment]
@adamghill 

so my workaround for this was to abstract out the 2nd level child (parent -> dashboard -> filter/list component) into a separate class. Then we just have one component... and use inheritance...

```
class DashboardEventsBrowserView(FindEventsViewCommon, UnicornView):
```

this sucks

I do not understand why my component works fine when used directly from main view, but not from another component? I am not doing any thing fancy IMO - I pass a list of ids to the component and show some dropdowns.

**Comment by adamghill** [comment]
I think this should be fixed by 0.58.0 of `Unicorn` which drastically improves child components. Feel free to re-open this if it still is a problem, though.
